### PR TITLE
Replace List.chop with List.splitAt and remove its own List.take from illib

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
Also, remove `List.take` from `illib.fs`

### CODE REVIEW

**Comment by forki** [comment]
❤️

File: src/fsharp/SignatureConformance.fs [hunk 1]
```diff
@@ -236,7 +236,7 @@ type Checker(g, amap, denv, remapInfo: SignatureRepackageInfo, checkingSig) =
                   err(fun(x, y, z) -> FSComp.SR.ValueNotContainedMutabilityGenericParametersDiffer(x, y, z, string mtps, string ntps))
                 elif implValInfo.KindsOfTypars <> sigValInfo.KindsOfTypars then
                   err(FSComp.SR.ValueNotContainedMutabilityGenericParametersAreDifferentKinds)
-                elif not (sigArgInfos.Length <= implArgInfos.Length && List.forall2 (fun x y -> List.length x <= List.length y) sigArgInfos (fst (List.chop sigArgInfos.Length implArgInfos))) then 
+                elif not (sigArgInfos.Length <= implArgInfos.Length && List.forall2 (fun x y -> List.length x <= List.length y) sigArgInfos (fst (List.splitAt sigArgInfos.Length implArgInfos))) then
```
Comment (forki) [inline src/fsharp/SignatureConformance.fs]: splitat and fst -> could be changed to List.take right?

Comment (vasily-kirichenko) [reply src/fsharp/SignatureConformance.fs]: Yep
```fsharp
> let xs = [1..10];;
val xs : int list = [1; 2; 3; 4; 5; 6; 7; 8; 9; 10]

> List.splitAt 3 xs |> fst;;
val it : int list = [1; 2; 3]

> List.take 3 xs;;
val it : int list = [1; 2; 3]
```

Comment (vasily-kirichenko) [reply src/fsharp/SignatureConformance.fs]: Done.

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Ubuntu16.04 Release_default Build please

**Comment by forki** [comment]
@dotnet-bot test Ubuntu16.04 Release_default Build please

**Comment by dsyme** [comment]
@dotnet-bot test Ubuntu16.04 Release_default Build please

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Ubuntu16.04 Release_default Build please

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Ubuntu16.04 Release_default Build please

**Comment by vasily-kirichenko** [comment]
Oh my, illib's `take` does not throw if called with out of bound index, so we cannot just remove it :(

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Ubuntu16.04 Release_default Build please

**Pull request merged**